packages/core/src/typeschema/validation.ts

```diff
@@ -225,9 +225,17 @@ class ResourceValidator {
       return;
     }
 
+    const structureIssue = createStructureIssue(path, 'Invalid ' + type + ' format');
```

Comment:
There's a tradeoff here between "duplicate code" and unnecessary allocations.  This will always execute the code path of allocating memory for the issue even if it's not needed.  I'd suggest that we remove this line, and tolerate the duplicate `this.issues.push(createStructureIssue(path, 'Invalid ' + type + ' format'));` to avoid unnecessary execution.